tasks/in_place_upgrade.yml

```diff
@@ -0,0 +1,53 @@
+# get version numbers from all jar files found
+- name: "in_place_upgrade : detect version to upgrade from."
+  set_fact:
+     aem_cms_in_place_upgrade_existing_version: "{{ aem_cms_existing_quickstarts.files | map(attribute='path') | map('regex_replace', '^.*cq-quickstart-(.+)-standalone-quickstart.jar$', '\\1') | list | sort | reverse | list | first}}"
+
+# safe gate which only allows upgrades when allowed
+- name: "in_place_upgrade : fail when upgrade is not allowed."
+  fail:
+    msg:
+      - "In place upgrade is not allowed, 'aem_cms_in_place_upgrade' is set to '{{ aem_cms_in_place_upgrade }}'"
+      - "existing version: '{{ aem_cms_in_place_upgrade_existing_version }}'"
+      - "version to deploy: '{{ aem_cms_version }}'"
+  when: aem_cms_in_place_upgrade != true
+
+# fail on downgrade
+- name: "in_place_upgrade : fail on downgrade."
+  fail:
+    msg:
+    - "Downgrading is not allowed/supported!"
+    - "existing version: '{{ aem_cms_in_place_upgrade_existing_version }}'"
+    - "version to deploy: '{{ aem_cms_version }}'"
+  when: aem_cms_version | regex_replace('\\D','') | int < aem_cms_in_place_upgrade_existing_version | regex_replace('\\D','') | int
+
+# check if upgrading is allowed based on existing and new version
+- name: "in_place_upgrade : fail when upgrade path from '{{ aem_cms_in_place_upgrade_existing_version }}' to '{{ aem_cms_version }}' is not allowed."
+  fail:
+    msg: "Upgrade from {{ aem_cms_in_place_upgrade_existing_version }} to {{ aem_cms_version }} is not allowed."
+  when: aem_cms_in_place_upgrade_paths[aem_cms_in_place_upgrade_existing_version] is not defined or
+        aem_cms_version not in aem_cms_in_place_upgrade_paths[aem_cms_in_place_upgrade_existing_version]
+
+# ensure instance is stopped before renaming the jars
+- name: "in_place_upgrade : stopping existing instance for upgrade."
+  service:
+    name: "{{ aem_cms_service_name }}"
+    state: stopped
+
+# rename old quickstart jars to *.jar.bak because default start script will use first jar found in the app folder
```

Comment:
Directly moving the file (i.e. `command: mv ..`) is an option as it would be a bit faster and more compact

Reply:
That is correct, but this would not be the ansible way to do it.

Reply:
Had a second thought about it and i will change it to "mv". This will speed the thing up and saves space.